vault.py

```diff
@@ -134,16 +151,30 @@ def run(self, terms, inject=None, variables=None, **kwargs):
 
         return [result['data'][field]] if field is not None else [result['data']]
 
+    def _fetch_approle_token(self, cafile, capath, role_id, secret_id, approle_role_path, url, cahostverify):
+        request_url = urljoin(url, approle_role_path)
+        req_params = {
+            'role_id': role_id,
+            'secret_id': secret_id
+        }
+        result = self.__fetch_client_token(cafile, capath, request_url, req_params, cahostverify)
+        token = result['auth']['client_token']
+        return token
+
     def _fetch_token(self, cafile, capath, github_token, url, cahostverify):
```

Comment:
I like the `_fetch_approle_token` you've added above, please rename this method to `_fetch_github_token`